rclpy/test/test_destruction.py

```diff
@@ -144,11 +143,7 @@ def test_destroy_subscription_asap():
             with sub.handle:
                 pass
 
-            with sub.handle:
-                node.destroy_subscription(sub)
-                # handle valid because it's still being used
-                with sub.handle:
-                    pass
+            node.destroy_subscription(sub)
```

Comment:
same question applies to the similar changes in this file and others

Reply:
It was changed to prevent a case where destruction could be blocked forever. 
There's a longer description in the body of ros2/rclpy#739. In short, the subscription is still valid to existing users, but no new users are allowed. This prevents new users extending an object's lifetime indefinitely.

Reply:
mmm, I don't understand exactly, I will have to look at the code a bit more.

What does happen with the old code snippet now?
Will it throw an InvalidHandle exception? Will it block?

Reply:
I would expect the inner `with sub.handle` to raise `InvalidHandle`.

Reply:
Okay, I understand the change now.
Do we still the `Handle` class? It seems to have been replaced with shared ptrs and Destroyable (I'm not sure if completely though).

There's a minimal chance this breaks pre-existing code, but I don't think anybody was relying on this behavior.

Reply:
On `master` the `Handle` class is gone, but I'm not sure which PR to backport. The only use I see of it on this branch is in `test_handle.py`.